src/Command/ErrorFormatter/ErrorFormatterFactory.php

```diff
@@ -0,0 +1,21 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Command\ErrorFormatter;
+
+class ErrorFormatterFactory {
+
+	/** @var string */
+	private $formatType;
+
+	/** @var \Nette\DI\Container */
+	private $container;
+
+	public function __construct(string $formatType, \Nette\DI\Container $container) {
+		$this->formatType = $formatType;
+		$this->container = $container;
+	}
+
+	public function create() {
+		return $this->container->getService('errorFormatter.'.$this->formatType);
```

Comment:
I really don't like the dependency on the container here.  Is there some better way to achieve this with nette?  I couldn't figure out how to compile this information into the container itself.